apps/server/src/api-data/report/report.service.ts

```diff
@@ -0,0 +1,84 @@
+import { OntimeReport, OntimeEventReport } from 'ontime-types';
+import { RuntimeState } from '../../stores/runtimeState.js';
+
+//TODO: there seams to be some actions that should invalidate reports
+// events timer edits?
+// event delete
+// Also what about roll mode?
```

Comment:
I think there are too many edge cases here and we should keep it simple

The report, should tell the user how long did something take last time it ran
I dont think it should matter what playback mode it is in

As for clearing:
- The user would chose to clear the entire report
- The user deletes an event, so we can clear the report for it

Reply:
> I dont think it should matter what playback mode it is in

I agree we just have to find all the places to hook in.

as for the clearing
the way we display the report right now it is relative to the events duration, so of the user changes the duration of the event, dose the report still make sense?

Reply:
> I agree we just have to find all the places to hook in.

What if we use the same place where we do the integrations?
In that way, that becomes the side effects place

> the way we display the report right now it is relative to the events duration, so of the user changes the duration of the event, dose the report still make sense?

I believe the report should be a feature that says "last time this ran, this is what happened", in which case it doesnt matter if the user changes things post fact. This also has advantage of keeping the feature simpler and a more straightforward mental model